core/src/server.cpp

```diff
@@ -289,6 +283,27 @@ void Server::allowMessage()
     can_send_ic_messages = true;
 }
 
+void Server::handleDiscordIntegration()
+{
+    if (discord != nullptr) {
```

Comment:
Why not just add a reload method within discord instead of deleting the entire thing just to reconstruct it?

Reply:
That would require to include server to disconnect and connect the signals, if it happens in Discord.

My version ensures all ties are servered and I simply do not have to add the logic that is required to recheck all connections and disconnect them if necessary.

Additionally nothing inside of Discord has any  reason to be preserved as it's loaded from ConfigManager on demand.

TLDR : Work in my solution < Work required to handle signals and slots in server.

Reply:
Nothing in Discord requires it to be destroyed to work properly. As far as I can see, the only thing that's really accomplishing is letting it create that uptime timer. This can literally be done by a function within Discord. handleDiscordIntegration() can call this function, because server has a pointer to discord. There is no point deleting discord just to remake it for this one thing. You don't need to destroy all the signals and slots just to remake them all. If we're doing ban webhooks now when we weren't before, you connect it. If we're no longer doing them, you disconnect it. That's it. You don't have to delete it.

You have a pointer to discord with Server, there is no point to not using it.